File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.go
func minCostClimbingStairs(cost []int) int {
    take, notTake := cost[0], cost[1]
    for i:=2; i<len(cost); i++ {
        takeThis := cost[i] + min(take, notTake)
        take, notTake = notTake, takeThis
    }
    return min(take, notTake)
}

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.java
public class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int take = cost[0], notTake = cost[1];
        for(var i=2; i<cost.length; i++){
            var takeThis = cost[i] + Math.min(take, notTake);
            take = notTake;
            notTake = takeThis;
        }

        return Math.min(take, notTake);
    }
}